cmd/db_list.go

```diff
@@ -0,0 +1,77 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/anchore/grype/grype/db"
+	"github.com/spf13/cobra"
+)
+
+var dbListOutputFormat string
+
+var dbListCmd = &cobra.Command{
+	Use:   "list",
+	Short: "list all DBs available according to the listing URL",
+	Args:  cobra.ExactArgs(0),
+	RunE:  runDBListCmd,
+}
+
+func init() {
+	dbListCmd.Flags().StringVarP(&dbListOutputFormat, "output", "o", "text", "format to display results (available=[text, raw, json])")
+
+	dbCmd.AddCommand(dbListCmd)
+}
+
+func runDBListCmd(_ *cobra.Command, _ []string) error {
+	dbCurator, err := db.NewCurator(appConfig.DB.ToCuratorConfig())
+	if err != nil {
+		return err
+	}
+
+	listing, err := dbCurator.ListingFromURL()
+	if err != nil {
+		return err
+	}
+
+	supportedSchema := dbCurator.SupportedSchema()
+	available, exists := listing.Available[supportedSchema]
+
+	if len(available) == 0 || !exists {
+		fmt.Printf("No databases available for the current schema (%d)\n", supportedSchema)
+		return nil
+	}
+
+	switch dbListOutputFormat {
+	case "text":
+		// summarize each listing entry for the current DB schema
+		for _, l := range available {
+			fmt.Printf("Built:    %s\n", l.Built)
+			fmt.Printf("URL:      %s\n", l.URL)
+			fmt.Printf("Checksum: %s\n\n", l.Checksum)
+		}
+
+		fmt.Printf("%d databases available for schema %d\n", len(available), supportedSchema)
+	case "json":
+		// show entries for the current schema
+		enc := json.NewEncoder(os.Stdout)
+		enc.SetEscapeHTML(false)
+		enc.SetIndent("", " ")
+		if err := enc.Encode(&available); err != nil {
+			return fmt.Errorf("failed to db listing information: %+v", err)
+		}
+	case "raw":
+		// show the entire listing file
+		enc := json.NewEncoder(os.Stdout)
+		enc.SetEscapeHTML(false)
+		enc.SetIndent("", " ")
+		if err := enc.Encode(&listing); err != nil {
+			return fmt.Errorf("failed to db listing information: %+v", err)
+		}
```

Comment:
heads up — This may boil down to a naming concern.... When I saw the format's name and description:

>raw: the entire listing file from the db.update-url

I was expecting this to be a raw (i.e. never parsed) representation of the file content retrieved for the listing URL. I could see this being very useful as a utility command for folks wanting to obtain a raw listing file w/o using the `curl` approach or worrying about JSON data structure (or even that the listing file is still being represented as JSON).

My suggestion would be to do one of the following: (1) adjust the implementation to show the raw bytes of the listing file, or (2) rename the format option to something besides "raw", to match the existing implementation (e.g. "all").

Curious for your thoughts here!

Reply:
I see your point, however, after lots of thought I can't seem to think of a better name. This seems to be the best candidate so far (anyone have a better name?) and the description documents the specific behavior.

Reply:
Let's think more on this over time, and treat this as a non-blocking conversation!